### DISCUSSION TITLE
Release v1.0.0-rc1

### DISCUSSION THREAD

**Discussion by miurahr** [miurahr/py7zr#607] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/miurahr/py7zr/releases/tag/v1.0.0-rc1'>Release v1.0.0-rc1</a>.</em>

**Comment by miurahr** [comment]
`read` and `readall` methods make users to believe `py7zr.SevenZipFile` working like `zipfile.ZipFile` without the reason.
It is better to drop the methods to stop users to consume their time with wrong understanding and claiming invalid tickets.

I have a plan to make a breaking change in -rc2 release.

**Comment by miurahr** [comment]
There are so many feature changes after `v0.22.0` and `v1.0.0-rc1`, and hot discussion on read/readall methods. It may be better to release it as `v0.23.0` and more before going to `v1.0.0` release stabilize.

**Comment by aborel** [comment]
Hello,
I think I was one of these users who used this functionality based on an incorrect assumption. With your plan, will v1.0.0-rc2 and future versions provide the correct functionality (probably with different function calls), or do you have something different in mind?

**Reply by miurahr** [reply]
You are right. I don't have another idea for it.
